### ISSUE TITLE
meta: double join executor's parallelism

### ISSUE THREAD

**Issue by skyzh** [risingwavelabs/risingwave#3311]
Hummock is performing already good enough at a lot of aspects:

* scan -- 10MB/s throughput for 64KB block size, 40MB/s throughput for 1MB block size, single thread
* short scan latency -- 300ms max for 64KB block size, 100ms max for 1MB block size

I believe this cannot be optimized further as our storage service can at most provide such good performance. So we should optimize join executors.

Basically there are a lot of ways to optimize, e.g. I/O queue, message reordering, etc. But I think the most intuitive way is to simply allocate more parallelism for join and other stateful executors.

**Comment by skyzh** [comment]
After some experiments, even doubling the parallelism doesn’t work. The bottleneck might be in other places.
